### PR TITLE
Upgrade to Web3 1.7.4

### PR DESCRIPTION
**Pull request by cds-amal**
This PR upgrades to latest web3 (1.7.4)

It turns out Truffle had a Ganache teardown issue with the decoder tests because the test providers used `eager instamine mode`. Truffle should use `instamine strict` mode, as it models how GETH and other clients behave. A good summary of how the different modes behave is [outlined in this doc](https://hackmd.io/wv7c5wn5Q0ebUWUH-N1H5A)

### CODE REVIEW

File: packages/decoder/package.json [hunk 1]
```diff
@@ -21,7 +21,7 @@
     "build": "tsc",
     "prepare": "yarn build",
     "start": "tsc --watch",
-    "test": "mocha ./test/current/test/* ./test/legacy/test/*"
+    "test": "mocha ./test/current/test/* ./test/legacy/test/* --exit"
```
Comment (gnidan) [inline packages/decoder/package.json:24]: Can we make sure this isn't necessary when using the decoder as a library? (i.e., not via tests)

**Review by gnidan — DISMISSED** [review]
(Requesting changes until it's clear what's going on with those open listeners that we're observing with the decoder tests)

File: packages/decoder/test/current/test/compatible-nativize.js [hunk 2]
```diff
@@ -11,31 +10,34 @@ const Codec = require("@truffle/codec");
...
   let abstractions;
-  let compilations;
   let web3;
 
-  let Contracts;
-
   before("Create Provider", async function () {
     provider = Ganache.provider({
+      miner: {
+        blockTime: 1
```
Comment (cds-amal) [inline packages/decoder/test/current/test/compatible-nativize.js:22]: This should be replaced with 
`instamine: "strict"`

File: packages/decoder/package.json (hunk 1 above)
Comment (wbt) [reply packages/decoder/package.json]: Is there a missing `provider.disconnect()` somewhere in there?

Comment (cds-amal) [reply packages/decoder/package.json]: There were missing `await providerr.disconnect()` statements, as well as our tests using the incorrect Ganache instamine mode! Thanks to @MicaiahReid for the assist on tracking this down.

**Comment by haltman-at** [comment]
Note that in addition to the conflict listed here, this also needs to have `db-kit` updated now that #5192 is merged.

**Review by haltman-at — APPROVED** [review]

**Pull request merged**